packages/cli/test/integration/generators/repository.integration.js

```diff
@@ -0,0 +1,343 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const path = require('path');
+const assert = require('yeoman-assert');
+const testlab = require('@loopback/testlab');
+const fs = require('fs');
+
+const expect = testlab.expect;
+const TestSandbox = testlab.TestSandbox;
+
+const generator = path.join(__dirname, '../../../generators/repository');
+
+const testUtils = require('../../test-utils');
+
+// Test Sandbox
+const SANDBOX_PATH = path.resolve(__dirname, '..', '.sandbox');
+const sandbox = new TestSandbox(SANDBOX_PATH);
+
+// Sandbox fixture files definition
+const DATASOURCE_FIXTURES_PATH = 'datasources';
+const MODEL_FIXTURES_PATH = 'models';
+const DATASOURCE_APP_PATH = 'src/datasources';
+const MODEL_APP_PATH = 'src/models';
+const REPOSITORY_APP_PATH = 'src/repositories';
+const INDEX_FILE = path.join(SANDBOX_PATH, REPOSITORY_APP_PATH, 'index.ts');
+
+const SANDBOX_FILES = [
+  {
+    fixturePath: DATASOURCE_FIXTURES_PATH,
+    path: DATASOURCE_APP_PATH,
+    file: 'dbkv.datasource.json',
+    content: `
```

Comment:
Please don't use string literals to define JSON objects, use `JSON.stringify` instead.

```ts
content: JSON.stringify({
  name: "dbkv",
  // ...
});
```